File: cz/mg/vulkan/vk/PFNvkRegisterObjectsNVX.java
package cz.mg.vulkan.vk;

/**
 *  @see <a href="https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/PFN_vkRegisterObjectsNVX.html">khronos documentation</a>
 **/
public class PFNvkRegisterObjectsNVX extends VkFunctionPointer {
    public PFNvkRegisterObjectsNVX() {
    }

    public PFNvkRegisterObjectsNVX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkRegisterObjectsNVX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkRegisterObjectsNVX(long value) {
        setValue(value);
    }

    public PFNvkRegisterObjectsNVX(VkInstance instance) {
        super(instance, new VkString("vkRegisterObjectsNVX"));
    }

    public void call(VkDevice device, VkObjectTableNVX objectTable, VkUInt32 objectCount, VkObjectTableEntryNVX.Pointer ppObjectTableEntries, VkUInt32 pObjectIndices, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.NULL_ADDRESS, objectTable != null ? objectTable.getVkAddress() : VkPointer.NULL_ADDRESS, objectCount != null ? objectCount.getVkAddress() : VkPointer.NULL_ADDRESS, ppObjectTableEntries != null ? ppObjectTableEntries.getVkAddress() : VkPointer.NULL, pObjectIndices != null ? pObjectIndices.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.NULL_ADDRESS);
    }

    private static native void call(long vkaddress, long device, long objectTable, long objectCount, long ppObjectTableEntries, long pObjectIndices, long rval);
}

File: c/cz/mg/vulkan/PFN_vkRegisterObjectsNVX.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkRegisterObjectsNVX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong objectTable, jlong objectCount, jlong ppObjectTableEntries, jlong pObjectIndices, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkRegisterObjectsNVX f = (PFN_vkRegisterObjectsNVX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkObjectTableNVX*)jniLongToPointer(objectTable)),
        *((uint32_t*)jniLongToPointer(objectCount)),
        ((VkObjectTableEntryNVX**)jniLongToPointer(ppObjectTableEntries)),
        ((uint32_t*)jniLongToPointer(pObjectIndices))
    );
}
